### DISCUSSION TITLE
Is it possible to display it in a popupbox in WPF?

### DISCUSSION THREAD

**Discussion by VahidEra** [dotnet/wpf#7806] [Q&A] [answered]
Hi guys,
I am using material desighn component in my wpf project and I want to display a WindowsFormsHost in a popupbox, after doing this unfortunately the WindowsFormsHost is not displayed in the popupbox.
Is it possible to do this, or is it not possible do to the difference between WPF and Windows Form controls?

**Comment by miloush** [comment]
Do you have a demo project?

**Reply by VahidEra** [reply]
@miloush Dear Jan, this is my sample project:
https://drive.google.com/file/d/1cV7m-Y5DczyVHI8EV43lecaXzEbcZK_i/view?usp=sharing

**Reply by miloush** [reply, marked as answer]
What doesn't go together is WS_EX_LAYERED (a type of transparent window) and Win32 children windows (Winforms controls). In other words you can show WindowsFormsHost in a Popup, but it cannot be transparent in traditional WPF sense. You either disable transparency (`Popup.AllowsTransparecncy = False`), or, you use Win32 window rather than WPF window to host the popup, or, you use more complicated techniques like `SetLayeredWindowAttribute` API.

Unfortunately for you, the Material Design is relying on the transparency to make popups with blurred edges. To see what I mean, in your sample project you can do this:
```xaml
<md:PopupBox.Resources>
    <Style TargetType="md:PopupEx">
        <Style.Triggers>
            <EventTrigger RoutedEvent="Loaded">
                <EventTrigger.Actions>
                    <BeginStoryboard>
                        <Storyboard>
                            <BooleanAnimationUsingKeyFrames Storyboard.TargetProperty="AllowsTransparency">
                                <DiscreteBooleanKeyFrame Value="False" />
                            </BooleanAnimationUsingKeyFrames>
                        </Storyboard>
                    </BeginStoryboard>
                </EventTrigger.Actions>
            </EventTrigger>
        </Style.Triggers>
    </Style>
</md:PopupBox.Resources>
```
(it has to be done via animation to override the local value in the PopupBox template) and the Winforms controls will render, but you will see that there is now black border everwhere which is supposed to be semi-transparent.

I am afraid there is no easy way. The easiest solution I think would be to not use the Material Design popup, and use the built-in one. For example:
```xaml
<Grid>
    <Button Name="button" HorizontalAlignment="Left" Margin="280,174,0,173" Click="OnClick"/>
    <Popup Name="popup" PlacementTarget="{Binding ElementName=button}" Placement="Bottom">
        <WindowsFormsHost xmlns:wf="clr-namespace:System.Windows.Forms;assembly=System.Windows.Forms" Background="#FFDA1B1B" Width="100" Height="100">
            <wf:Label Text="this is a test."/>
        </WindowsFormsHost>
    </Popup>        
</Grid>
```
where `button` on click just sets `popup.IsOpen = true`.

**Reply by VahidEra** [reply]
@miloush Dear Jan, You are exactly right. Thank you very much for solving the problem and for your help This popup solves my problem and there is no need for a popupbox Thanks again for the clear and thorough guide and explanations that always solve my problems.  You are my teacher because everything I learned in WPF you taught me.
